slsim/cluster_lens.py

```diff
@@ -0,0 +1,714 @@
+import numpy as np
+from lenstronomy.Cosmo.lens_cosmo import LensCosmo
+from lenstronomy.Util import constants
+from lenstronomy.LensModel.lens_model import LensModel
+from lenstronomy.LightModel.light_model import LightModel
+from lenstronomy.LensModel.Solver.lens_equation_solver import LensEquationSolver
+from lenstronomy.LensModel.Solver.lens_equation_solver import (
+    analytical_lens_model_support,
+)
+from slsim.ParamDistributions.gaussian_mixture_model import GaussianMixtureModel
+from lenstronomy.Util import util, data_util
+from slsim.lensed_system_base import LensedSystemBase
+from slsim.lens import image_separation_from_positions
+import warnings
+
+
+class ClusterLens(LensedSystemBase):
```

Comment:
I've removed the ClusterLens class and implemented `NFWCluster` which is a new `Deflector` subclass to manage clusters. It can work with the existing `Lens` class by modifying the `deflector_mass_model_lenstronomy`. Another way might be to handle the mass model in the `Deflector` class and just call it from the `Lens` class (see #156).